packages/terra-application-layout/examples/index-examples/ExampleApplication.jsx

```diff
@@ -0,0 +1,289 @@
+  path: '/page_1',
+  text: 'Page 1',
+}, {
+  path: '/page_2',
+  text: 'Page 2',
+}, {
+  path: '/page_3',
+  text: 'Page 3',
+}, {
+  path: '/page_4',
+  text: 'Page 4',
+}, {
+  path: '/page_5',
+  text: 'Page 5',
+}, {
+  path: '/page_6',
+  text: 'Page 6',
+}, {
+  path: '/page_7',
+  text: 'Page 7',
+}];
+
+/**
+ * The indexPath will be given to the NavigationLayout to set up the appropriate redirects. If users attempt to navigate to a path unsupported
+ * by the routingConfig, they will be redirected to this route. This path should therefore be present in the routingConfig.
+ */
+const indexPath = '/page_1';
+
+const userAvatar = (
+  <Avatar
+    image={ProfilePicture}
+    variant="user"
+    alt="Swanson, Henry"
+    ariaLabel="Swanson, Henry"
+    key="user_avatar"
+  />
+);
+
+const userData = {
+  name: 'Swanson, Henry',
+  detail: 'Henry Swanson\'s my name, and excitement\'s my game.',
+  photo: userAvatar,
+};
+
+/**
+ * The data provided for nameConfig will be visible in the ApplicationLayout's header, as well
+ * as in any menus at the tiny and small breakpoints.
+ */
+const nameConfig = Object.freeze({
+  title: 'Example Application',
+  accessory: <Image variant="rounded" src="https://github.com/cerner/terra-core/raw/master/terra.png" height="26px" width="26px" />,
+});
+
+const customUtilityItems = [{
+  key: 'additional-1',
+  title: 'Drill-in Item',
+  childKeys: [
+    'additional-sub-1',
+    'additional-sub-2',
+  ],
+  parentKey: Utils.utilityHelpers.defaultKeys.MENU,
+}, {
+  key: 'additional-sub-1',
+  title: 'Addtional Item 1 - Sub 1',
+  parentKey: 'additional-1',
+}, {
+  key: 'additional-sub-2',
+  title: 'Addtional Item 1 - Sub 2',
+  parentKey: 'additional-1',
+}, {
+  key: 'additional-2',
+  contentLocation: Utils.utilityHelpers.locations.BODY,
+  title: 'Custom Checkbox Item',
+  isSelectable: true,
+  parentKey: Utils.utilityHelpers.defaultKeys.MENU,
+}, {
+  key: 'additional-3',
+  contentLocation: Utils.utilityHelpers.locations.FOOTER,
+  title: 'Custom Footer',
+  parentKey: Utils.utilityHelpers.defaultKeys.MENU,
+}];
+
+const ExampleApplication = withRouter(injectIntl(({ location, intl }) => {
+  /**
+   * The data provided for utilityConfig will be visible in the ApplicationLayout's header in the
+   * standard rendering mode and within the menus in the compact rendering mode.
+   *
+   * The ApplicationLayout's Utils export provides a helper function named getDefaultUtilityConfig that will
+   * generate the configuration for the standard set of utility options. If the standard configuration is not
+   * desirable, an entirely custom configuration can be used instead.
+   */
+  const utilityConfig = Object.freeze({
+    title: 'Swanson, Henry',
+    accessory: userAvatar,
+    menuItems: Utils.utilityHelpers.getDefaultUtilityItems(intl, userData, customUtilityItems),
+    initialSelectedKey: Utils.utilityHelpers.defaultKeys.MENU,
+    onChange: (event, itemData, disclose) => {
```

Comment:
We should also update a menu item's `isSelected` attribute, otherwise if an item has `isSelectable` set, its checkmark's state will not be correct